apps/sms/views/shared/js/activity_handler.js

```diff
@@ -353,7 +353,8 @@ var ActivityHandler = {
 
             // Close notification if we are already in thread view and view
             // become visible.
-            if (document.hidden && threadId === Threads.currentId) {
+            if (document.hidden &&
+              Navigation.isCurrentPanel('thread', { id: threadId })) {
```

Comment:
nit: please move the line 2 spaces to the right, so that "Navigation" is just below "document".